File: Swift/001TwoSum/Sources/TwoSum/TwoSum.swift
import Foundation

class TwoSum {
	func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
		guard nums.count > 1 else { return [] }

		var previouslyEncountered = [Int: Int]()
		for (idx, value) in nums.enumerated() {
			let diff = target - value
			if let exists = previouslyEncountered[diff] {
				return [exists, idx]
			} else {
				previouslyEncountered[value] = idx
			}
		}
		return []
	}
}

File: Python/001TwoSum/two_sum.py
from typing import *


def two_sum(nums: List[int], target: int) -> List[int]:
  seen_so_far = {}
  for idx, val in enumerate(nums):
    diff = target - val
    if diff in seen_so_far:
      return [seen_so_far[diff], idx]
    else:
      seen_so_far[val] = idx
  return []
